### ISSUE TITLE
<component/operator/ctk>: Enforce naming conventions for the microservices inside a component

### ISSUE THREAD

**Issue by abbeydom** [tmforum-oda/oda-component-ctk#21]
## Description

The ctk is to enforce the naming convention for the microservices inside a component. When components are deployed with the same release names, the microservices  may conflict. Therefore, ctk should enforce a naming convention for the microservices that includes the component name e.g.:
e.g. metadata: name: {{.release.name}}-{{.component.name}}-{{.microservices.name}}
i.e. r1-prodcatalogue-productcatalogapi

**Comment by abbeydom** [comment]
Is this still needed given recent updates to the ctk?

**Comment by LesterThomas** [comment]
> Is this still needed given recent updates to the ctk?

Yes - there is still an issue that you can't deploy multiple instances of the example component as the microservice names conflict (we had someone facing this issue just this morning!).
